scheduler/core/components/optimizer/greedymax.py

```diff
@@ -37,7 +37,7 @@ class GreedyMaxOptimizer(BaseOptimizer):
     """
 
     def __init__(self, min_visit_len: timedelta = timedelta(minutes=30), show_plots: bool = False):
-        # self.selection = Selection
+        self.selection = Selection
```

Comment:
This isn't quite right, since you're setting `self.selection` to a type (`Selection`). It would be like doing:

```
self.index = int
```

Instead, what we should do is:
```
self.selection: Optional[Selection] = None
```
(saying that `self.selection` will be a `Selection`, but right now, we don't have the value for it).

Then when you set it in `setup`, it looks good.